File: Greedy/122. Best Time to Buy and Sell Stock II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0
        ans = 0
        for i in range(len(prices) - 1):
            if prices[i+1] > prices[i]:
                ans += prices[i+1] - prices[i]
        return ans

File: Greedy/122. Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = size(prices);
        if (n == 0)
            return n;
        int count = 0;
        for (int i = 1; i < n; ++i){
            if (prices[i] > prices[i-1]){
                count = count + prices[i] - prices[i-1];
            }
        }
        return count;
    }
};